internal/provider/resource_gitlab_project_milestone_test.go

```diff
@@ -0,0 +1,97 @@
+//go:build acceptance
+// +build acceptance
+
+package provider
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccGitlabProjectMilestone_basic(t *testing.T) {
+
+	rInt1, rInt2 := acctest.RandInt(), acctest.RandInt()
+	project := testAccCreateProject(t)
+
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckGitlabProjectMilestoneDestroy,
+		Steps: []resource.TestStep{
+			{
+				// create Milestone with required values only
+				Config: fmt.Sprintf(`
+                resource "gitlab_project_milestone" "this" {
+                  project = "%v"
+                  title   = "test-%d"
+                }`, project.PathWithNamespace, rInt1),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("gitlab_project_milestone.this", "iid"),
+					resource.TestCheckResourceAttrSet("gitlab_project_milestone.this", "milestone_id"),
+					resource.TestCheckResourceAttrSet("gitlab_project_milestone.this", "updated_at"),
+					resource.TestCheckResourceAttrSet("gitlab_project_milestone.this", "created_at"),
+					resource.TestCheckResourceAttrSet("gitlab_project_milestone.this", "web_url"),
+					resource.TestCheckResourceAttrSet("gitlab_project_milestone.this", "expired"),
+				),
+			},
+			{
+				// verify import
+				ResourceName:      "gitlab_project_milestone.this",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				// update some Milestone attributes
+				Config: fmt.Sprintf(`
+                resource "gitlab_project_milestone" "this" {
+                  project     = "%[1]d"
+                  title       = "test-%[2]d"
+                  description = "test-%[2]d"
+                  start_date  = "2022-04-10"
+                  due_date    = "2022-04-15"
+                  state       = "closed"
+                }`, project.ID, rInt2),
```

Comment:
NIT: for better readability:

```suggestion
				Config: fmt.Sprintf(`
                                 resource "gitlab_project_milestone" "this" {
                                   project     = "%[1]d"
                                   title       = "test-%[2]d"
                                   description = "test-%[2]d"
                                   start_date  = "2022-04-10"
                                   due_date    = "2022-04-15"
                                   state       = "closed"
                                 }`, project.ID, rInt2),
```